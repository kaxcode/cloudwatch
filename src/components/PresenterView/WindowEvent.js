import React from 'react';
import PropTypes from 'prop-types';

class WindowEvent extends React.PureComponent {
  static propTypes = {
    event: PropTypes.string.isRequired,
    handler: PropTypes.func.isRequired
  };

  componentDidMount() {
    this.addEvent();
  }

  componentWillUpdate() {
    this.removeEvent();
  }

  componentDidUpdate() {
    this.addEvent();
  }

  componentWillUnmount() {
    this.removeEvent();
  }

  render() {
    return null;
  }

  addEvent() {
    const { event, handler } = this.props;
    window.addEventListener(event, handler);
  }

  removeEvent() {
    const { event, handler } = this.props;
    window.removeEventListener(event, handler);
  }
}

export default WindowEvent;
