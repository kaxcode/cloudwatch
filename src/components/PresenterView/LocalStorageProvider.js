import React from 'react';
import { arrayOf, string, func } from 'prop-types';
import WindowEvent from './WindowEvent.js';

class LocalStorageProvider extends React.Component {
  static propTypes = {
    keys: arrayOf(string),
    children: func
  };
  state = {
    someState: {
      timeRemaining: 0
    }
  };

  componentDidMount = () => {
    this.setState({ timeRemaining: localStorage.timeRemaining });
  };
  handleStorage = e => {
    const someState = this.props.keys.reduce((set, key) => {
      set[key] = localStorage.getItem(key);
      console.log(set);
      return set;
    }, {});
    this.setState({ someState });
    console.log(e);
    console.log(someState);
  };
  render() {
    return (
      <React.Fragment>
        <WindowEvent event="storage" handler={this.handleStorage} />
        {this.props.children(this.state.someState)}
      </React.Fragment>
    );
  }
}

export default LocalStorageProvider;
