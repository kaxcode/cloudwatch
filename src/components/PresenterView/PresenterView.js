import React from 'react';
import LocalStorageProvider from './LocalStorageProvider.js';
import Timer from '../Timer/Timer.js';

class PresenterView extends React.Component {
  render() {
    return (
      <LocalStorageProvider keys={['timeRemaining']}>
        {({ timeRemaining }) => <Timer timeRemaining={timeRemaining} />}
      </LocalStorageProvider>
    );
  }
}

export default PresenterView;
