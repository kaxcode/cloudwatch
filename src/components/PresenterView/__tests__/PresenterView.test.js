import React from 'react';
import Enzyme from 'enzyme';
import PresenterView from '../PresenterView';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });
jest.useFakeTimers();

describe('Stopwatch', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<PresenterView />);
  });

  it('renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
