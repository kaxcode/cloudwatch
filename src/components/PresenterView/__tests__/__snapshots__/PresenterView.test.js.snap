// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stopwatch renders correctly 1`] = `
<LocalStorageProvider
  keys={
    Array [
      "timeRemaining",
    ]
  }
/>
`;
